File: algos simple/insertionSort.java
public class insertionSort
{
    public static void insertSortInc(int[] arr)
    {
        for (int i = 0; i < arr.length-1; i++)
        {
            int key = arr[i+1];
            int j;
            for (j = i; j >= 0 && arr[j] > key; j--)
            {
                arr[j+1] = arr[j];
            }
            arr[j+1] = key;
        }
    }
    public static void insertSortDec(int[] arr)
    {
        for (int i = 0; i < arr.length-1; i++)
        {
            int key = arr[i+1];
            int j;
            for (j = i; j >= 0 && arr[j] < key; j--)
            {
                arr[j+1] = arr[j];
            }
            arr[j+1] = key;
        }

    }

    static void printArray(int[] arr)
    {
        for (int i : arr)
        {
            System.out.print(i + " ");
        }
        System.out.println();
    }

    public static void main(String[] args)
    {
        int[] arr = {12, 11, 13, 5, 6};
        insertSortInc(arr);
        printArray(arr);
        insertSortDec(arr);
        printArray(arr);
    }
}

File: algos simple/insertionSort.cpp
#include <iostream>
#include <vector>
using namespace std;

void insertionSort(vector<int>& arr, int n){

    for (int i = 1; i < n; i++)
    {
        int key = arr[i];
        int j = i-1;

        while (j>=0 && key < arr[j])
        {
             arr[j+1] = arr[j];
             j--;
        }
        arr[j+1] = key;
    }
}

void printArray(vector<int>& arr)
{
    for (int i : arr)
    {
        cout << i << " ";
    }
}

int main(){
    vector<int> arr = {5, 11, 4, 7, 9};

    insertionSort(arr, 5);
    printArray(arr);

}